proto/spire/api/types/federationrelationship.proto

```diff
@@ -0,0 +1,39 @@
+syntax = "proto3";
+package spire.api.types;
+option go_package = "github.com/spiffe/spire-api-sdk/proto/spire/api/types";
+
+message FederationRelationship {
+    // The trust domain name (e.g., "example.org") to federate with. 
+    string trust_domain = 1;
+
+    // URL of the SPIFFE bundle endpoint that provides the trust
+    // bundle to federate with. Must use the HTTPS protocol.
+    string bundle_endpoint_url = 2;
+
+    // Required. The endpoint profile type.
+    oneof bundle_endpoint_profile {
+        // Use Web PKI endpoint profile. 
+        WebPKI https_web = 3;
+
+        // Use SPIFFE Authentication endpoint profile.
+        SPIFFEAuthentication https_spiffe = 4;
+    }
+}
+
+message SPIFFEAuthentication {
```

Comment:
Should we also have a bundle included in this message?

If I am creating a new relationship with SPIFFE auth ... I need to set the bundle first. Do we need to call bundle api, set the bundle ... _then_ call this api to set up the relationship?

I think this starts to tickle some of my earlier comments about understanding this resource a little better. It is closely related with the Bundles resource ... I do think there are use cases for bundles without federation, for example if you roll your own bundle distribution mechanism. But the relationship makes these API boundaries a little less clear. Curious to hear your thoughts on this

Reply:
This is a good point. I was hesitant to include the bundle here but I was influenced by the current implementation where you are forced to set the federated bundles separately from the configuration of the federation relationships.
I think that having the possibility to include the bundle here will make things easier and avoid extra calls to set federated bundles separately, mostly in the "bootstrapping of federation" process.
I would make this optional so if you don't provide it, it looks for a bundle previously stored.

Reply:
Specifying the bundle here makes a lot of sense in the case of chaining trust with SPIFFE Authentication, because you need the bundle of the trust domain where the bundle endpoint server resides and you may not need/have a federation relationship with that trust domain.